compiler.go

```diff
@@ -890,6 +922,187 @@ found:
 	s.bindings = s.bindings[:l]
 }
 
+func (c *compiler) compileModule(module *SourceTextModuleRecord) {
```

Comment:
I did intend on merging [compileModule](https://github.com/mstoykov/goja/blob/5f81441173723ebb181873312ffaa98604cdd2b0/compiler.go#L925) with `compile` but it seems like it might be a bit more involved, but there is quite a bit of repetition - so I would likely at least try unless you are against it. 

I don't think I will try to be adding `RunModule` a kin to `RunScript` as that still requires quite a lot of changes.